src/symmetric_heap_c.c

```diff
@@ -396,3 +399,27 @@ void SHMEM_FUNCTION_ATTRIBUTES * shmemalign(size_t alignment, size_t size)
 {
     return shmem_align(alignment, size);
 }
+
+
+void SHMEM_FUNCTION_ATTRIBUTES *
+shmemx_malloc_with_hints(size_t size, long hints)
+{
+    void *ret = NULL;
+
+    SHMEM_ERR_CHECK_INITIALIZED();
+
+    if (size == 0) return ret;
+
+    // Check for valid hints. If hint is invalid fall back to shmem_malloc and raise a debug warning
+    if(hints >= SHMEMX_MALLOC_MAX_HINTS && shmem_internal_my_pe == 0) {
+        DEBUG_STR("Invalid hint for shmem_malloc_with_hints. Valid hints are 0, SHMEMX_MALLOC_ATOMICS_REMOTE and SHMEMX_MALLOC_SIGNAL_REMOTE");
```

Comment:
Instead of RAISE_WARN_STR I've used DEBUG_STR to clear up the output. Let me know what you guys @jdinan @davidozog @wrrobin think of this

Reply:
The macro will print the function name, file, and line number. Rather than enumerating the valid hints, I would tend to prefer something like:
```suggestion
        DEBUG_MSG("Invalid hint: 0x%lx", (unsigned long) hints);
```
But also, this is an application error so DEBUG doesn't seem right here because SOS will silently ignore the bad hints unless the user sets the SHMEM_DEBUG env var. I think I would prefer RAISE_WARN_MSG. I also would not mask the warning based on PE ID, in case the user also failed to provide the same hints at every PE. At some point, when the hints are actually used, this will become a more serious error.

Reply:
Just to confirm my understanding, in case of bad hints, according to the definition of malloc_with_hints in spec SOS is supposed to ignore it, right? with RAISE_WARN_MSG the output gets filled with long warnings and it will be difficult to debug for users if multiple hints are  used unless we print the invalid hint with the warning. So i'll change it back to RAISE_WARN_STR and will also add the invalid hint in the output. Does that sound good?

Reply:
I agree about the PE id condition. This issue had come up in my discussion with @davidozog . But I still added it thinking that as of now it won't be an issue and we can change it later when we add more functionality/hints to malloc_with_hints. If we're adding RAISE_WARN_STR then removing this condition will make the output too long for users to debug.

Reply:
This is what the spec has to say with regard to the hints argument: "The valid usage hints are described in Table 4. Multiple hints may be requested by combining them with a bitwise OR operation. A zero option can be given if no options are requested."

I suppose it is a matter of interpretation what to do if the hints are not all valid?

Reply:
yeah that's true.It's kind of unclear in case of invalid hints . So I'll change this to RAISE_WARN_MSG and will remove the PE id condition so we don't have to worry about that in future.